byron/ledger/executable-spec/src/Ledger/Core.hs

```diff
@@ -460,15 +466,32 @@ toSet = Set.fromList . toList
 deriveGoblin ''Addr
 deriveGoblin ''BlockCount
 deriveGoblin ''Epoch
-deriveGoblin ''Hash
-deriveGoblin ''Lovelace
 deriveGoblin ''Owner
 deriveGoblin ''Sig
 deriveGoblin ''Slot
 deriveGoblin ''SlotCount
 deriveGoblin ''VKey
 deriveGoblin ''VKeyGenesis
 
+instance GeneOps g => Goblin g Hash where
+  tinker gen
+    = tinkerRummagedOrConjureOrSave
+        ((\x -> Hash (modulate x))
+           <$$> tinker ((\(Hash x) -> x) <$> gen))
+  conjure = saveInBagOfTricks =<< (Hash . modulate <$>
+    conjure)
+
+modulate :: Integral a => a -> a
```

Comment:
If we don't use `modulate` elsewhere we might want to make this a local definition. Although I would't mind just using the `rhs`, i.e. 

```
Hash . (`mod` 30) <$> conjure
```